src/entity/Entity.php

```diff
@@ -658,6 +658,18 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 			$ev = new EntityDamageEvent($this, EntityDamageEvent::CAUSE_VOID, 10);
 			$this->attack($ev);
 			$hasUpdate = true;
+		}else{
+			$entityBlock = $this->getWorld()->getBlock($this->getLocation());
+			$entityBox = $this->getBoundingBox()->expandedCopy(0.001, 0.1, 0.001);
+			foreach (Facing::ALL as $face) {
+				$block = $entityBlock->getSide($face);
+				foreach ($block->getCollisionBoxes() as $blockBox) {
+					if ($entityBox->intersectsWith($blockBox)) {
+						$block->onEntityCollide($this, Facing::opposite($face));
```

Comment:
should this case add this?
```php 
$hasUpdate = true;
```

Reply:
May be but if i realy good understand, `hasUpdate` variable is set to true only if entity is update. However, in function `onCollide`, we are not sure that entity is update.
So, i don't realy sure

Reply:
Normally we add `bool` returns to these kinds of hooks so that the caller knows whether or not something happened.